src/components/Shared/AddressTab.js

```diff
@@ -163,20 +159,20 @@ const AddressTab = ({
                 { key: 'flName', value: 'Foreign Language Name' }
               ]}
               values={addressValidation.values}
-              required
+              required={!requiredOptional}
               onChange={(event, newValue) => {
-                setCityStore([])
-                addressValidation.setFieldValue('stateId', null)
-                addressValidation.setFieldValue('cityId', null)
-                addressValidation.setFieldValue('city', null)
-                addressValidation.setFieldValue('cityDistrictId', null)
-                addressValidation.setFieldValue('cityDistrict', null)
-
+                addressValidation.setFieldValue('stateId', 0)
```

Comment:
you can't fill stateId with 0 when clearing the country or you will get an foreign key error when saving without a state because there is no stateId = 0 in the database
this is what I got: 
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_SYAB_SYST". The conflict occurred in database "R1003", table "dbo.SYST", column 'recordId'. The statement has been terminated.

send it as empty string instead

Reply:
ok done